portfolio/src/main/java/com/google/sps/servlets/NewMapMarkerServlet.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.search.checkers.GeoPointChecker;
+import com.google.maps.GeoApiContext;
+import com.google.maps.GeocodingApi;
+import com.google.maps.model.AddressComponent;
+import com.google.maps.model.AddressComponentType;
+import com.google.maps.model.GeocodingResult;
+import com.google.sps.data.MapMarker;
+import lombok.SneakyThrows;
+
+// Servlet that creates a map marker
+@WebServlet("/new-map-marker")
+public class NewMapMarkerServlet extends HttpServlet {
+
+  private final GeoApiContext geoApiContext = new GeoApiContext.Builder()
+    .apiKey("AIzaSyCfSokI5WnEOJZg12yfV-mINDnFaClsj6M").build();
+   
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    MapMarker newMapMarker = new MapMarker(
+      request.getParameter("collegeName"),
+      request.getParameter("internName"), 
+      getLongitude(request.getParameter("collegeAddress")),
+      getLatitude(request.getParameter("collegeAddress")));
+      
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(newMapMarker.makeEntity());
+
+    response.sendRedirect("/index.html");
+  }
+  
+  @SneakyThrows
+  public long getLongitude(String address) {
+    GeocodingResult[] results = GeocodingApi.geocode(geoApiContext, address).await();
+    GeocodingResult result = results[0];
```

Comment:
You should check the size of "result" and throw a NotFoundException if it is empty. Just assuming that it will have at least one element is dangerous.